man/river-status.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH river-status 1  "river-status 0.2.6" 
.SH NAME
river\-status \- A status information client for river
.SH SYNOPSIS
\fBriver\-status\fR [\fB\-a\fR|\fB\-\-all\fR] [\fB\-f\fR|\fB\-\-focused\fR] [\fB\-l\fR|\fB\-\-layout\fR] [\fB\-m\fR|\fB\-\-mode\fR] [\fB\-o\fR|\fB\-\-output\fR] [\fB\-p\fR|\fB\-\-pretty\fR] [\fB\-s\fR|\fB\-\-seat\fR] [\fB\-\-show\-outputs\fR] [\fB\-\-show\-seat\fR] [\fB\-\-sleep\fR] [\fB\-T\fR|\fB\-\-tag\fR] [\fB\-t\fR|\fB\-\-title\fR] [\fB\-u\fR|\fB\-\-urgent\fR] [\fB\-\-view\-tags\fR] [\fB\-w\fR|\fB\-\-watch\fR] [\fB\-h\fR|\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] 
.SH DESCRIPTION
river\-status is a CLI tool that displays information about running instancies of river.
The output is always in JSON, but you can select to have it pretty printed for human readablility.
.PP
By default, it will display the names of each output and the current seat river is using.
.PP
NOTE: this tool uses the `river\-status\-unstable\-v1` protocol which might be subject to significant breaking changes before river sees a stable 1.0 release.
.SH OPTIONS
.TP
\fB\-a\fR, \fB\-\-all\fR
Equivalent of \-f \-l \-m \-t \-\-tag \-u \-\-view\-tags
.TP
\fB\-f\fR, \fB\-\-focused\fR
Print information about the focused tags
.TP
\fB\-l\fR, \fB\-\-layout\fR
Print layout name of the focused tags.
.TP
\fB\-m\fR, \fB\-\-mode\fR
Print mode name.
.TP
\fB\-o\fR, \fB\-\-output\fR \fI<OUTPUT>\fR
Select the output to display information about.
.TP
\fB\-p\fR, \fB\-\-pretty\fR
Pretty print JSON.
.TP
\fB\-s\fR, \fB\-\-seat\fR \fI<SEAT>\fR
Select the seat to display information about.
.TP
\fB\-\-show\-outputs\fR
Explicitly print outputs even if no other flag connected to outputs is in use.
.TP
\fB\-\-show\-seat\fR
Explicitly print seat even if no other flag connected to seat is in use.
.TP
\fB\-\-sleep\fR
optional delay (in milliseconds) between calls to river for status updates. This option is a no\-op without `\-\-watch`.
.TP
\fB\-T\fR, \fB\-\-tag\fR
Output the key *focusedTags* and numerical value representing which tag is focused for each output.
.TP
\fB\-t\fR, \fB\-\-title\fR
Print the title of the focused view.
.TP
\fB\-u\fR, \fB\-\-urgent\fR
Print information about urgent tags if there are any.
.TP
\fB\-\-view\-tags\fR
Prints the tags of all views.
.TP
\fB\-w\fR, \fB\-\-watch\fR
Continuously print information as it changes.
.TP
\fB\-h\fR, \fB\-\-help\fR
Print help (see a summary with \*(Aq\-h\*(Aq)
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version
.SH VERSION
v0.2.6
.SH AUTHORS
Rickard Hedlund <@>


CHANGELOG
# Changelog

## v0.2.6 (2025-12-29)
- dependency bump
- follow the new branch pattern of https://github.com/riverwm/river.git and lock it to `0.3.x`
- lint `clippy::string_to_string` has been removed and been replaced by `clippy:implicit_clone`
- bump `rustc` support to 1.92.0 (ded5c06cf 2025-12-08) from `rustc` 1.90.0 (1159e78c4 2025-09-14)

## v0.2.5 (2025-09-26)
- dependency bump

## v0.2.4 (2025-09-05)
- dependency bump

## v0.2.3 (2025-08-16)
- dependency bump

## v0.2.2 (2025-06-18)
- dependency bump
- changed from using `once_cell::sync::Lazy` to `std::sync::LazyLock` (https://rust-lang.github.io/rust-clippy/master/index.html#non_std_lazy_statics)

## v0.2.1 (2025-05-07)
- dependency bump

## v0.2.0 (2025-03-17)

- rethink and remove `--no-seat` and `--no-output`. Instead of allowing the user to remove some parts of the output,
  allow the user to select what to view with `--show-outputs` and `--show-seat`.
  If the user selects any flag that is connected to output, `--show-outputs` will be implicit.
  If the user selects any flag that is connected to seat, `--show-seat` will be implicit.

## v0.1.3 (2025-03-15)

- BUGFIX: the option `-p` and `--pretty` does not pretty print the output.

## v0.1.2 (2025-03-14)

- add option `--sleep` as a way to limit number of calls per second to river for status updates
- apply code linting standards from https://github.com/dotfive/standards-rs#code-linting

## v0.1.1 (2025-03-10)

- add options `--no-seat` and `--no-output`
- standardize output to camelCase
- added forgotten option `--mode`
- move layout to be a part of output

## v0.1.0 (2025-03-01)
**If anyone uses this, all constructive feedback is useful.**

- First version


completions/bash/river-status.bash
_river-status() {
    local i cur prev opts cmd
    COMPREPLY=()
    if [[ "${BASH_VERSINFO[0]}" -ge 4 ]]; then
        cur="$2"
    else
        cur="${COMP_WORDS[COMP_CWORD]}"
    fi
    prev="$3"
    cmd=""
    opts=""

    for i in "${COMP_WORDS[@]:0:COMP_CWORD}"
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="river__status"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        river__status)
            opts="-a -f -l -m -o -p -s -T -t -u -w -h -V --all --focused --layout --mode --output --pretty --seat --show-outputs --show-seat --sleep --tag --title --urgent --view-tags --watch --help --version"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --output)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --seat)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

if [[ "${BASH_VERSINFO[0]}" -eq 4 && "${BASH_VERSINFO[1]}" -ge 4 || "${BASH_VERSINFO[0]}" -gt 4 ]]; then
    complete -F _river-status -o nosort -o bashdefault -o default river-status
else
    complete -F _river-status -o bashdefault -o default river-status
fi
